awx/main/tasks.py

```diff
@@ -1272,10 +1273,22 @@ def run(self, pk, **kwargs):
                     params.pop('inventory'),
                     os.path.join(private_data_dir, 'inventory')
                 )
+                pod_manager = None
```

Comment:
@shanemcd what if we moved this _entire_ block up ^^^

Reply:
..to the *very* beginning of the method *before* we do `status = running`:

https://github.com/ansible/awx/pull/4189/files#diff-9d4ea1dd908b35fb92eaede4bd10bb46R1146

We could wrap it with a `try...except...` and if we encountered an exception (like a 403 for resource quota problems), we could log it, and then just return (and leave the job in pending state).

Reply:
e.g.,

```python
def run(...):
    if containerized:
        try:
            launch_k8s()
        except 403:
            logger.warning('out of quota')
            return # job status in pending
        except Exception:
            self.update_model(pk, status='error')
            logger.exception('k8s error happened')
            return
```

Reply:
cc @kdelee

Reply:
`except 403`  <--- obviously pseudo-code

I dunno what actual exception you all were encountering, might need to ask @one-t so we can reproduce and catch it.

Reply:
@shanemcd also, to clarify further, I don't think we should do *any* of this in *this* PR (it's my suggestion for https://github.com/ansible/awx/issues/4910)

Reply:
We have a patch that solves this but are holding in until after initial merge so we can run it through the full regression suite.